src/metric/rank_metric.cc

```diff
@@ -541,20 +561,31 @@ struct EvalAucPR : public Metric {
   bst_float Eval(const HostDeviceVector<bst_float> &preds,
                  const MetaInfo &info,
                  bool distributed) override {
+    CHECK_NE(info.labels_.Size(), 0U) << "label set cannot be empty";
+    CHECK_EQ(preds.Size(), info.labels_.Size())
+        << "label size predict size not match";
+    std::vector<unsigned> tgptr(2, 0);
```

Comment:
Seems a simple array is more suitable.

Reply:
it is assigned to a group (from metainfo), if present. hence, a compatible type is used here.